ios/SailthruMobile.framework/Headers/STMAttributes.h

```diff
@@ -15,10 +15,87 @@ typedef NS_ENUM(NSInteger, STMAttributesMergeRule) {
 
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wdeprecated-declarations"
-@interface STMAttributes : CarnivalAttributes
-#pragma clang diagnostic pop
+@interface STMAttributes : NSObject
+
+/**
+ *  Sets a string object for a given key. The key must not be nil.
+ *
+ *  @param string The string to set.
+ *  @param key The key to set.
+ */
+- (void)setString:(nullable NSString *)string forKey:(nonnull NSString *)key;
+
+/**
+ *  Sets an array of string objects for a given key. All items in the array must be of type NSString. The key must not be nil.
+ *
+ *  @param strings The array of strings to set.
+ *  @param key The key to set.
+ */
+- (void)setStrings:(nullable NSArray<NSString *> *)strings forKey:(nonnull NSString *)key;
+
+/**
+ *  Sets a date objects for a given key. The key must not be nil.
+ *
+ *  @param date The date to set.
+ *  @param key The key to set.
+ */
+- (void)setDate:(nullable NSDate *)date forKey:(nonnull NSString *)key;
+
+/**
+ *  Sets an array of date objects for a given key. All items in the array must be of type NSDate. The key must not be nil.
+ *
+ *  @param dates The array of dates to set.
+ *  @param key The key to set.
+ */
+- (void)setDates:(nullable NSArray<NSDate *> *)dates forKey:(nonnull NSString *)key;
+
+/**
+ *  Sets an integer object for a given key. The key must not be nil.
+ *
+ *  @param integer The integer to set.
+ *  @param key The key to set.
+ */
+- (void)setInteger:(NSInteger)integer forKey:(nonnull NSString *)key;
+
+
+/**
+ *  Sets an array of integers objects for a given key. All items in the array must be of type NSNumber (constructed with an NSInteger). The key must not be nil.
+ *
+ *  @param integers The array of integers to set.
+ *  @param key The key to set.
+ */
+- (void)setIntegers:(nullable NSArray<NSNumber *> *)integers forKey:(nonnull NSString *)key;
+
+/**
+ *  Sets a boolean object for a given key. The key must not be nil.
+ *
+ *  @param aBool The boolean to set.
+ *  @param key The key to set.
+ */
+- (void)setBool:(BOOL)aBool forKey:(nonnull NSString *)key;
+
+/**
+ *  Sets a float object for a given key. The key must not be nil.
+ *
+ *  @param aFloat The float to set.
+ *  @param key The key to set.
+ */
+- (void)setFloat:(CGFloat)aFloat forKey:(nonnull NSString *)key;
+
+/**
+ *  Sets an array of floats objects for a given key. All items in the array must be of type NSNumber (constructed with a CGFloat). The key must not be nil.
+ *
+ *  @param floats The array of floats to set.
+ *  @param key The key to set.
+ */
+- (void)setFloats:(nullable NSArray<NSNumber *> *)floats forKey:(nonnull NSString *)key;
+
+/**
+ *  Sets the method of merging on the attributes. STMAttributesMergeRuleUpdate will merge with existing attributes and STMAttributesMergeRuleReplace will replace all data with the new set.
+ *
+ *  @param mergeRule The STMAttributesMergeRule to use. The default is update.
+ */
+- (void)setAttributesMergeRule:(STMAttributesMergeRule)mergeRule;
 
```

Comment:
Sorry another silly question! Just noticed there are 2x `CarnivalAttributes` files but only 1 `STMAttributes` file (similar for a few other files too) - have we just combined the code from these 2 Carnival files into 1 Sailthru?

Reply:
There were two versions of the CarnivalAttributes header because it was present in both Carnival.framework and SailthruMobile.framework, and both these frameworks were available in the last release. Now Carnival.framework has been removed and the CarnivalAttributes.h file has been removed from SailthruMobile.framework. Before the rebrand there was only 1 version and after change it will go back to just one version but of the rebrand instead.

Reply:
Awesome thanks for clarifying Ian!